polymorphia-backend/src/main/java/com/agh/polymorphia_backend/repository/user/UserRepository.java

```diff
@@ -2,11 +2,23 @@
 
 import com.agh.polymorphia_backend.model.user.User;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
 import java.util.Optional;
 
 @Repository
 public interface UserRepository extends JpaRepository<User, Long> {
     Optional<User> findByEmail(String email);
+
+    @Modifying
+    @Query("""
+        DELETE FROM User u 
+        WHERE u.password IS NULL 
+        OR u.email IN :expiredTokenEmails
+    """)
+    Integer deleteUsersWithoutPasswordOrExpiredTokens(@Param("expiredTokenEmails") List<String> expiredTokenEmails);
```

Comment:
NIT: Currently it looks to me like if we wanted to delete users without password or without expired tokens
```suggestion
    Integer deleteUsersWithoutPasswordOrWithExpiredTokens(@Param("expiredTokenEmails") List<String> expiredTokenEmails);
```